chain/client-primitives/src/types.rs

```diff
@@ -339,8 +339,48 @@ pub struct Status {
     pub is_health_check: bool,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum StatusError {
+    #[error("IO Error: {error_message}")]
+    IOError { error_message: String },
```

Comment:
I think it does not help us much and we may just merge it into the InternalError variant. (I think it is fine to have just IOError, but once we have InternalError on the list, it feels like we have two variants representing the same thing, and IOError does not even record the original error)